nominal/core/datasource.py

```diff
@@ -111,6 +113,66 @@ def get_channels(
             )
             yield from (Channel._from_channel_metadata_api(self._clients, channel) for channel in response.responses)
 
+    def get_write_stream(
+        self,
+        batch_size: int = 50_000,
+        max_wait: timedelta = timedelta(seconds=1),
+        data_format: Literal["json", "protobuf"] = "json",
+    ) -> WriteStream:
+        """Stream to write non-blocking messages to a datasource.
```

Comment:
while we're here..."non-blocking messages" is a weird misnomer

```suggestion
        """Stream to write messages to a datasource.
        
        Messages are written asynchronously.
```